alf/environments/adv_fetch_envs/adv_fetch_env.py

```diff
@@ -0,0 +1,118 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+
+from gym.envs.robotics.fetch_env import FetchEnv
+from gym.envs.robotics import robot_env, rotations, utils
+
+
+class AdvFetchEnv(FetchEnv):
+    def __init__(
+            self,
+            model_path,
+            n_substeps,
+            gripper_extra_height,
+            block_gripper,
+            has_object,
+            target_in_the_air,
+            target_offset,
+            obj_range,
+            target_range,
+            distance_threshold,
+            initial_qpos,
+            reward_type,
+    ):
+        """Initializes a new Fetch environment. Almost the same with ``FetchEnv``.
+        # The only change is from 4 to 7 for ``n_actions`` where the extra 3 dims
+        represent 'xyz' euler angles rotated compared to the previous step.
+
+        .. note::
+
+            After adding the rotation angles, the robot gripper might more easily
+            have collisions with the table or itself. Maybe in the future we should
+            have the environment return collision info.
+
+            You can overwrite the function `_sample_goal()` to define new goals
+            given these extra rotation actions, also `_get_obs()` to redefine the
+            achieved goal.
+
+        Args:
+            model_path (string): path to the environments XML file
+            n_substeps (int): number of substeps the simulation runs on every call to step
+            gripper_extra_height (float): additional height above the table when positioning the gripper
+            block_gripper (boolean): whether or not the gripper is blocked (i.e. not movable) or not
+            has_object (boolean): whether or not the environment has an object
+            target_in_the_air (boolean): whether or not the target should be in the air above the table or on the table surface
+            target_offset (float or array with 3 elements): offset of the target
+            obj_range (float): range of a uniform distribution for sampling initial object positions
+            target_range (float): range of a uniform distribution for sampling a target
+            distance_threshold (float): the threshold after which a goal is considered achieved
+            initial_qpos (dict): a dictionary of joint names and values that define the initial configuration
+            reward_type ('sparse' or 'dense'): the reward type, i.e. sparse or dense
+        """
+        self.gripper_extra_height = gripper_extra_height
+        self.block_gripper = block_gripper
+        self.has_object = has_object
+        self.target_in_the_air = target_in_the_air
+        self.target_offset = target_offset
+        self.obj_range = obj_range
+        self.target_range = target_range
+        self.distance_threshold = distance_threshold
+        self.reward_type = reward_type
+        self._quat = None
+
+        robot_env.RobotEnv.__init__(
```

Comment:
`super().__init__(...)`

Reply:
> `super().__init__(...)`

I actually need to call its grandparent's init.

Reply:
> `super().__init__(...)`

I actually need to call its grandparent's init.